hack-remove-global-override.service

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Remove hack flatpak global override
+DefaultDependencies=no
+Conflicts=shutdown.target
+Wants=local-fs.target
+After=local-fs.target
+
+# Only run on updates
+Before=multi-user.target systemd-update-done.service
+ConditionNeedsUpdate=|/etc
+ConditionNeedsUpdate=|/var
+
+ConditionPathExists=/var/lib/flatpak/overrides/global
```

Comment:
Looking at [the hook](https://github.com/endlessm/eos-image-builder/blob/eos3.6/hooks/image/51-hack-flatpak-overrides.chroot), on Hack systems this file will be a symbolic link into the HackComponents app. What happens if, on a current hack system, you run `pkexec flatpak override --talk-name=com.example.DummyName`? I can think of 3 possibilities:

1. The operation fails because the symlink points to a file which is read-only
2. The operation succeeds, and replaces the symlink with a file
3. The operation succeeds, and modifies the file inside the HackComponents app, corrupting the Flatpak repo

I really hope it's 1 or 2!

If 1, what we have here is probably fine. If 2 or (gulp) 3, we might consider whether we want to handle this case differently and just remove the overrides that come from HackComponents, in case users have used global overrides? Or is that considered to be vanishingly unlikely?

systemd does support:

```ini
ConditionPathIsSymbolicLink=/var/lib/flatpak/overrides/global
```

but I'm not sure if that helps.

Reply:
I've just tested and the final result is the number 2, the symlink is replaced with a file with the same content and the new override.

I think we've two options here:

1. Remove the file only if it's a symbolic link, as you said, and in that case, if the users has modified the global override, the new clubhouse won't work correctly, because that override will replace the one in apps and we expect that new apps will have the `com.hack_computer.Clippy.Extension` not the old one.

2. The other option is to keep as is right now and it'll be removed so the user will loose their custom changes.

I think that the best option is to remove it and if the user has changed that (it should be a rare user case, it's not usual to override global flatpaks), he will loose the changes and should do again, but the new clubhouse and the old apps will continue working with the local overrides created by the clubhouse during the migration quest.

Reply:
I think you're right that it will be quite rare that a Hack user will have applied a system-wide global Flatpak override.